src/services/user/UserServiceContext.tsx

```diff
@@ -1,69 +0,0 @@
-"use client";
-import React, {
-  createContext,
-  useContext,
-  useState,
-  useEffect,
-  useMemo,
-} from "react";
-import {
-  createDefaultUserService,
-  UserService,
-} from "@/services/user/UserService";
-import { UserData } from "@/model/UserData";
-import { useRouter } from "next/navigation";
-import { Highscore } from "@/model/HighscoresEnum";
-
-const UserServiceContext = createContext<UserService | null>(null);
-const UserDataContext = createContext<UserData>({
-  achievements: {},
-  quizzes: {},
-  username: "",
-  highscores: { 0: 0 },
-});
-
-export function UserDataProvider({ children }: any) {
-  const router = useRouter();
-  const [userData, setUserData] = useState<UserData>({
-    achievements: {},
-    quizzes: {},
-    username: "",
-    highscores: { 0: 0 },
-  });
-  const userService = useMemo(() => createDefaultUserService(), []);
-
-  userService.onUpdate = (userData: UserData) => {
-    // copy the object, otherwise no update will be triggered
-    setUserData({ ...userData });
-  };
-
-  useEffect(() => {
```

Comment:
since you removed this callback that checks on every page reload if a user exists in the localstorage and if not redirects to "/" you have to replace the functionality that checks if a userid and gameCode object exits in the local storage and if not redirect to "/"